src/System.Runtime.InteropServices.RuntimeInformation/src/System/Runtime/InteropServices/RuntimeInformation/RuntimeInformation.Unix.cs

```diff
@@ -67,7 +71,11 @@ public static Architecture ProcessArchitecture
                 {
                     if (null == s_processArch)
                     {
-                        if (Architecture.Arm == OSArchitecture)
+                        if (Architecture.Arm64 == OSArchitecture)
+                        {
+                            s_processArch = s_is64BitProcess ? Architecture.Arm64 : Architecture.Arm;
```

Comment:
The `s_is64BitProcess` check does not make sense here. It will be always `true` given how things are implemented.

We should really have two different helpers in System.Native.so: One that returns the OSArchitecture and one that returns ProcessArchitecture. Same pattern as what we do on Windows. 

The implementation of these two helpers in System.Native.so will look the same for the currently supported platforms, but it will be prepared for situation where it can differ.